src/Assignments/ProblemDetails.tsx

```diff
@@ -12,16 +12,23 @@ const INFINITE_MAX_ATTEMPT_VALUE = 0;
 interface ProblemDetailsProps {
     problem: ProblemObject;
     topic: TopicObject | null;
+    attemptsRemaining?: number;
 }
 
 export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
     problem,
     topic,
+    attemptsRemaining,
 }) => {
-    const startDate = moment(topic?.startDate);
-    const endDate = moment(topic?.endDate);
-    const deadDate = moment(topic?.deadDate);
-    const solutionsMoment = moment(deadDate).add(1, 'days');
+    const isAssessment = !_.isNil(topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0]);
+    const isClosed = isAssessment && topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].isClosed;
+    const versionStartTime = topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].startTime;
+    const versionEndTime = topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].endTime;
+
+    const startDate = (_.isNil(versionStartTime)) ? moment(topic?.startDate) : moment(versionStartTime);
```

Comment:
This could be simplified by doing:
```TypeScript
const startDate = versionStartTime?.toMoment() ?? topic?.startDate.toMoment()
```

Although I guess the toMoment extension is only in the backend it will be in the frontend with my PR

The problem with the way it is now is if topic is undefined startDate is inaccurately the current time

We can change that later